### PR TITLE
User can join org

### PR DESCRIPTION
**Pull request by 0us**
Implement join organization from app 
 - ClubDetailedFragment now houses another fragment, contextually changing based
    on the users membership status 

 - Implement new API call: getMembership 
 - getMembership and joinOrg both return an array of the users memberships in the org

- Fix bug with password serialization: Password can now only be deserialized, and never serialized

### CODE REVIEW

File: app/src/androidTest/java/no/ntnu/klubbhuset/ExampleInstrumentedTest.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import android.content.Context;
```
Comment (markusranda) [inline app/src/androidTest/java/no/ntnu/klubbhuset/ExampleInstrumentedTest.java]: Do we need this example class?

File: app/src/androidTest/java/no/ntnu/klubbhuset/ui/userviews/club/ClubsViewModelTest.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package no.ntnu.klubbhuset.ui.userviews.club;
```
Comment (markusranda) [inline app/src/androidTest/java/no/ntnu/klubbhuset/ui/userviews/club/ClubsViewModelTest.java]: If this is a testclass, maybe it should be placed in the test package?

"IE303911-Klubbhuset/app/src/test/java/no/ntnu/klubbhuset"

File: server/src/main/java/no/ntnu/klubbhuset/domain/User.java [hunk 3]
```diff
@@ -32,10 +35,13 @@ public void addSecurityGroup(SecurityGroup securityGroup) {
     @Id
     private String email;
 
-    @JsonIgnore
-    @JsonbTransient
     private String password;
 
+    @JsonbTransient
```
Comment (markusranda) [inline server/src/main/java/no/ntnu/klubbhuset/domain/User.java]: Just curious, when you place the annotation "@JsonbTransient" on a method, does that inable the jsonbuilder serializing the password?

File: server/docker-compose.yml [hunk 4]
```diff
@@ -27,11 +27,11 @@ services:
       # Name of database
       MYSQL_DATABASE: klubbhuset
       # So you don't have to use root
-      MYSQL_USER: user
+      MYSQL_USER: lars
       # User password
-      MYSQL_PASSWORD: password
+      MYSQL_PASSWORD: ganymedes
```
Comment (markusranda) [inline server/docker-compose.yml]: Please change your weak password

File: app/src/main/java/no/ntnu/klubbhuset/ui/userviews/club/detailed/ClubDetailedNotMember.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
+package no.ntnu.klubbhuset.ui.userviews.club.detailed;
```
Comment (markusranda) [inline app/src/main/java/no/ntnu/klubbhuset/ui/userviews/club/detailed/ClubDetailedNotMember.java]: For consistency shouldn't every fragment-class include "fragment" at the end of the name?

File: app/src/main/java/no/ntnu/klubbhuset/ui/userviews/club/detailed/ClubDetailedMember.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
+package no.ntnu.klubbhuset.ui.userviews.club.detailed;
```
Comment (markusranda) [inline app/src/main/java/no/ntnu/klubbhuset/ui/userviews/club/detailed/ClubDetailedMember.java]: For consistency shouldn't every fragment-class include "fragment" at the end of the name?

File: server/src/main/java/no/ntnu/klubbhuset/domain/User.java (hunk 3 above)
Comment (spiftire) [reply server/src/main/java/no/ntnu/klubbhuset/domain/User.java]: Think so

File: app/src/androidTest/java/no/ntnu/klubbhuset/ui/userviews/club/ClubsViewModelTest.java (hunk 2 above)
Comment (0us) [reply app/src/androidTest/java/no/ntnu/klubbhuset/ui/userviews/club/ClubsViewModelTest.java]: Sjå igjen, vegen til klassa er "app/src/androidTest/java/no/ntnu/klubbhuset/ui/userviews/club/ClubsViewModelTest.java "

File: app/src/androidTest/java/no/ntnu/klubbhuset/ExampleInstrumentedTest.java (hunk 1 above)
Comment (0us) [reply app/src/androidTest/java/no/ntnu/klubbhuset/ExampleInstrumentedTest.java]: Fint å ha for referanse.

File: server/src/main/java/no/ntnu/klubbhuset/domain/User.java (hunk 3 above)
Comment (0us) [reply server/src/main/java/no/ntnu/klubbhuset/domain/User.java]: Ja

File: server/docker-compose.yml (hunk 4 above)
Comment (0us) [reply server/docker-compose.yml]: faen

File: app/src/main/res/layout/content_club_detailed.xml [hunk 7]
```diff
@@ -67,24 +67,28 @@
...
+        android:layout_marginBottom="16dp"
+        app:layout_constraintBottom_toTopOf="@+id/scrollView3"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/frameLayout" />
+        app:layout_constraintTop_toBottomOf="@+id/frameLayout"
+        tools:layout="@layout/content_club_detailed_not_member" />
 
     <ScrollView
+        android:id="@+id/scrollView3"
```
Comment (spiftire) [inline app/src/main/res/layout/content_club_detailed.xml]: Change id to be more descriptive

File: app/src/main/res/layout/content_club_detailed.xml [hunk 8]
```diff
@@ -103,6 +107,13 @@
                 android:layout_marginBottom="12dp"
                 android:background="?android:attr/listDivider" />
 
+            <TextView
+                android:id="@+id/textView7"
```
Comment (spiftire) [inline app/src/main/res/layout/content_club_detailed.xml]: Same as above

File: app/src/main/res/layout/content_club_detailed_member.xml [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:orientation="vertical"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent">
+
+        <TextView
+            android:id="@+id/textView8"
```
Comment (spiftire) [inline app/src/main/res/layout/content_club_detailed_member.xml]: Change id name

File: server/src/main/java/no/ntnu/klubbhuset/domain/User.java [hunk 10]
```diff
@@ -32,10 +35,13 @@ public void addSecurityGroup(SecurityGroup securityGroup) {
     @Id
     private String email;
 
-    @JsonIgnore
```
Comment (spiftire) [inline server/src/main/java/no/ntnu/klubbhuset/domain/User.java]: Wont this expose the password in the token?

File: app/src/main/java/no/ntnu/klubbhuset/ui/userviews/club/detailed/ClubDetailedNotMember.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (0us) [reply app/src/main/java/no/ntnu/klubbhuset/ui/userviews/club/detailed/ClubDetailedNotMember.java]: Poenget med disse fragmentene er å nettop unggå dette. Disse fragmentene (ClubDetailedNotMember, ClubDetailedMember) inneholder kun komponenter som relaterer til
tilstanden (medlem eller ikke medlem), og blir byttet ut kontekstuelt etter denne tilstanden. 

Se: 
bilde

File: server/src/main/java/no/ntnu/klubbhuset/domain/User.java (hunk 10 above)
Comment (0us) [reply server/src/main/java/no/ntnu/klubbhuset/domain/User.java]: JWT leser vel kun info fra usersecurityroles tabellen?

File: app/src/main/java/no/ntnu/klubbhuset/ui/userviews/club/detailed/ClubDetailedNotMember.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (spiftire) [reply app/src/main/java/no/ntnu/klubbhuset/ui/userviews/club/detailed/ClubDetailedNotMember.java]: Ok. Det gir mening. Tenkte bare vi kunne kjøre sjekker rett i fragmentet og forandre innholdet der etter, men er kanskje like ryddig på måten du har tenkt

**Pull request merged**